sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -124,14 +124,17 @@ case class BroadcastExchangeExec(
     case _ => 512000000
   }
 
+  @transient
+  private lazy val jobTag = s"broadcast exchange (runId ${runId.toString})"
+
   @transient
   override lazy val relationFuture: Future[broadcast.Broadcast[Any]] = {
     SQLExecution.withThreadLocalCaptured[broadcast.Broadcast[Any]](
       session, BroadcastExchangeExec.executionContext) {
           try {
-            // Setup a job group here so later it may get cancelled by groupId if necessary.
-            sparkContext.setJobGroup(runId.toString, s"broadcast exchange (runId $runId)",
-              interruptOnCancel = true)
+            // Setup a job tag here so later it may get cancelled by tag if necessary.
+            sparkContext.addJobTag(jobTag)
+            sparkContext.setInterruptOnCancel(true)
```

Comment:
cc @ulysses-you FYI. Now we can cancel the jobs in boradcast like this.

Reply:
Glad to see we support cancel job with tag!